comms/core/src/connectivity/peer_discovery_bridge.rs

```diff
@@ -0,0 +1,239 @@
+                available_peers
+            );
+            return Ok(false);
+        }
+
+        // Time-based throttling
+        if let Some(last_attempt) = self.last_discovery_attempt {
+            let min_interval = self.config.success_rate_tracking_window; // Use tracking window as discovery interval
+            if last_attempt.elapsed() < min_interval {
+                trace!(
+                    target: LOG_TARGET,
+                    "({}) Discovery throttled, last attempt {} ago (min interval: {})",
+                    task_id,
+                    format_duration(last_attempt.elapsed()),
+                    format_duration(min_interval)
+                );
+                return Ok(false);
+            }
+        }
+
+        info!(
+            target: LOG_TARGET,
+            "({}) Peer discovery needed: available_peers={}, discovery_threshold={}",
+            task_id,
+            available_peers,
+            discovery_threshold
+        );
+
+        Ok(true)
+    }
+
+    /// Execute peer discovery to find more peer candidates
+    pub async fn execute_discovery_with_pool(
+        &mut self,
+        task_id: u64,
+        connected_node_ids: &[crate::peer_manager::NodeId],
+    ) -> Result<usize, ConnectivityError> {
+        debug!(
+            target: LOG_TARGET,
+            "({}) Executing peer discovery to find more candidates",
+            task_id
+        );
+
+        self.last_discovery_attempt = Some(Instant::now());
+        proactive_dialing_metrics::increment_peer_discovery_attempts();
+
+        // Get current peer count for comparison (excluding already connected peers)
+        let initial_count = self.get_available_peer_count_with_pool(connected_node_ids).await?;
+
+        // For now, implement a basic peer discovery strategy
+        // This could be enhanced to integrate with DHT discovery, seed peer querying, etc.
+        let discovered_count = self.basic_peer_discovery(task_id).await?;
+
+        let final_count = self.get_available_peer_count_with_pool(connected_node_ids).await?;
+        let net_discovered = final_count.saturating_sub(initial_count);
+
+        // Update metrics
+        proactive_dialing_metrics::increment_peer_discovery_peers_found(net_discovered);
+
+        info!(
+            target: LOG_TARGET,
+            "({}) Peer discovery completed: attempted={}, net_discovered={}, total_available={}",
+            task_id,
+            discovered_count,
+            net_discovered,
+            final_count
+        );
+
+        Ok(net_discovered)
+    }
+
+    /// Get the current count of available (non-banned, non-connected) peer candidates
+    pub async fn get_available_peer_count_with_pool(
+        &self,
+        connected_node_ids: &[crate::peer_manager::NodeId],
+    ) -> Result<usize, ConnectivityError> {
+        let all_peers = self
+            .peer_manager
+            .all(None)
+            .await
+            .map_err(ConnectivityError::PeerManagerError)?;
```

Comment:
```suggestion
        let all_peers = self
            .peer_manager
            .discovery_syncing(...)
            .await
            .map_err(ConnectivityError::PeerManagerError)?;
```
We should not load up the entire database in memory and do filtering in rust, there are SQL functions to do this fo us.